File: libnavigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/BaseTest.java
package com.mapbox.services.android.navigation.ui.v5;

import com.mapbox.navigation.base.route.model.Route;
import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;

import java.io.IOException;

public class BaseTest {

  protected static final double DELTA = 1E-10;
  protected static final String ACCESS_TOKEN = "pk.XXX";

  private TestRouteBuilder routeBuilder;
  private TestRouteProgressBuilder routeProgressBuilder;

  public BaseTest() {
    routeBuilder = new TestRouteBuilder();
    routeProgressBuilder = new TestRouteProgressBuilder();
  }

  protected String loadJsonFixture(String filename) throws IOException {
    return routeBuilder.loadJsonFixture(filename);
  }

  protected Route buildTestDirectionsRoute() throws IOException {
    return routeBuilder.buildTestDirectionsRoute(null);
  }

  protected RouteProgress buildRouteProgress(Route route,
                                             double stepDistanceRemaining,
                                             double legDistanceRemaining,
                                             double distanceRemaining,
                                             int stepIndex,
                                             int legIndex) throws Exception {
    return routeProgressBuilder.buildTestRouteProgress(
      route, stepDistanceRemaining, legDistanceRemaining, distanceRemaining, stepIndex, legIndex
    );
  }
}

File: libtrip-notification/src/test/java/com/mapbox/navigation/trip/notification/BaseTest.kt
package com.mapbox.navigation.trip.notification

import com.mapbox.navigation.base.model.route.Route
import com.mapbox.navigation.base.model.route.RouteProgress
import java.io.IOException

open class BaseTest {

    companion object {
        @JvmStatic
        val DELTA = 1E-10
        @JvmStatic
        val LARGE_DELTA = 0.1
        @JvmStatic
        val ACCESS_TOKEN = "pk.XXX"
    }

    private val routeBuilder: TestRouteBuilder = TestRouteBuilder()
    private val routeProgressBuilder: TestRouteProgressBuilder = TestRouteProgressBuilder()

    @Throws(IOException::class)
    fun loadJsonFixture(filename: String): String {
        return routeBuilder.loadJsonFixture(filename)
    }

    @Throws(IOException::class)
    fun buildTestDirectionsRoute(): Route {
        return routeBuilder.buildTestDirectionsRoute(null)
    }

    @Throws(IOException::class)
    fun buildTestDirectionsRoute(fixtureName: String?): Route {
        return routeBuilder.buildTestDirectionsRoute(fixtureName)
    }

    @Throws(Exception::class)
    fun buildDefaultTestRouteProgress(): RouteProgress {
        val testRoute = routeBuilder.buildTestDirectionsRoute(null)
        return routeProgressBuilder.buildDefaultTestRouteProgress(testRoute)
    }

    @Throws(Exception::class)
    fun buildTestRouteProgress(
        route: Route,
        stepDistanceRemaining: Double,
        legDistanceRemaining: Double,
        distanceRemaining: Double,
        stepIndex: Int,
        legIndex: Int
    ): RouteProgress {
        return routeProgressBuilder.buildTestRouteProgress(
                route, stepDistanceRemaining, legDistanceRemaining, distanceRemaining, stepIndex, legIndex
        )
    }
}
